pkg/controller/identity/iamrole/controller.go

```diff
@@ -67,7 +73,7 @@ type connector struct {
 }
 
 func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (managed.ExternalClient, error) {
```

Comment:
It would be preferable to move away from using the `util.RetrieveAwsConfigFromProvider` function here I think, as has been done on other `v1beta1` objects: https://github.com/crossplane/provider-aws/blob/1d51592300403eea476170ac4f1b549cf1ae42a6/pkg/controller/database/rdsinstance.go#L91

However, I do think all of this authentication machinery will be refactored in the not too distant future so I would be fine if this didn't change for now.

Reply:
As with note on client, we can implement this in a future PR.